pages/index.tsx

```diff
@@ -1,19 +1,36 @@
 import { Box, Button, Container, Typography } from '@mui/material';
 import jwt from 'jsonwebtoken';
 import { NextPage } from 'next';
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import Cookies from 'universal-cookie';
 
 import createAxiosInstance from 'client/lib/axios';
 import { getDiscordOauthUrl } from 'client/lib/oauth';
 import { AUTH_COOKIE_NAME } from 'shared/constants';
 
+interface User {
+  id: number;
+  discord_name: string;
+}
```

Comment:
We've talked about interface's not being necessary in some cases. It's unclear to me if it's needed here.

Reply:
This should probably also be a shared type between the server and the client so that we can reference the "client shape User" both:
* when we make the call for a user on the client, and
* when we return it on the server